pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -389,6 +403,9 @@ func (k *Probe) getMapStats(stats *model.EBPFStats) error {
 	// Allow the maps to be garbage collected
 	k.mapBuffers.resetBuffers()
 
+	// Close unused programs in the prog helper cache
+	k.mphCache.closeUnusedPrograms()
```

Comment:
Could you collect the still present map IDs from the loop above and then pass them in here?

Reply:
done in the latest push

Reply:
I was thinking creating a slice of IDs in this function and then passing it, like this:   `k.mphCache.closeUnusedPrograms(liveMapIDs)` rather than adding multiple functions to the cache API.

Reply:
The disadvantage in doing that is that you maintain the slice even if the cache is not enabled (if the helper is not available)

Reply:
hmm, good point.